dss_datamover/socket_communication.py

```diff
@@ -306,68 +308,69 @@ def recv_json(self, format="JSON", timeout=RECV_TIMEOUT):
         :timeout: default 60 seconds
         :return: Return received data in json format.
         """
+        self.client_socket.settimeout(timeout)
         msg_len = None
         msg = "{}"
+        time_started = datetime.now()
+
         try:
+            # first receive message length from payload
             msg_len_in_bytes = b''
-            received_msg_len_size = 0
-            time_started = datetime.now()
-            # Iterate till we receive 10 bytes or timeout.
-            while received_msg_len_size < MESSAGE_LENGTH:
-                received_msg_len_in_bytes = self.client_socket.recv(MESSAGE_LENGTH - received_msg_len_size)
-                received_msg_len_size += len(received_msg_len_in_bytes)
-                msg_len_in_bytes += received_msg_len_in_bytes
-                time_spent_in_seconds = (datetime.now() - time_started).seconds
-                if time_spent_in_seconds >= timeout:
-                    raise socket.timeout("ServerSocket: Timeout ({} seconds) from recv function".format(time_spent_in_seconds))
-
-            if msg_len_in_bytes != b'':
-                msg_len = int(msg_len_in_bytes.decode('utf8'))
-        except socket.timeout as e:
-            raise e
-        except socket.error as e:
-            self.logger.error("ServerSocket: Determine msg length - {}".format(e))
-        except ValueError as e:
-            raise socket.error("ServerSocket: ValueError - {}".format(e))
-        except Exception as e:
-            self.logger.error("ServerSocket: {}".format(e))
+            msg_len_in_bytes = self.client_socket.recv(MESSAGE_LENGTH)
+            msg_len = int(msg_len_in_bytes.decode('utf8'))
+            if len(msg_len_in_bytes) != MESSAGE_LENGTH:
```

Comment:
msg_len variable to be used

Reply:
example 
msg_len = 131
msg_len_in_bytes = 0000000131
MESSAGE_LENGTH = 10